connector/connect/common/src/main/protobuf/spark/connect/expressions.proto

```diff
@@ -319,3 +320,14 @@ message PythonUDF {
   bytes command = 3;
 }
 
+message ScalarScalaUDF {
+  // (Required) Serialized JVM object containing UDF definition, input encoders and output encoder
+  bytes payload = 1;
+  // (Optional) Input type(s) of the UDF
+  repeated DataType inputTypes = 2;
+  // (Required) Output type of the UDF
+  DataType outputType = 3;
```

Comment:
@xinrong-meng @vicennial can we homogenize this and put the return type in the `CommonInlineUserDefinedFunction` message?